File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/aop/introduction/Stub.kt
package io.micronaut.docs.aop.introduction

// tag::imports[]

import io.micronaut.aop.Introduction
import io.micronaut.context.annotation.Bean
import io.micronaut.context.annotation.Type

import java.lang.annotation.Documented
import java.lang.annotation.Retention

import java.lang.annotation.RetentionPolicy.RUNTIME

// end::imports[]

// tag::class[]
@Introduction // <1>
@Type(StubIntroduction::class) // <2>
@Bean // <3>
@Documented
@Retention(RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class Stub(val value: String = "")
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/aop/introduction/Stub.py
# tag::imports[]
from micronaut.aop import Introduction
from micronaut.context.annotation import Bean
# end::imports[]

# tag::class[]
@Introduction # <1>
@Bean # <2>
def Stub(value : str = ""):
    def decorator(func):
        return func
    return decorator
# end::class[]
